### PR TITLE
Improve location for get_file_spec errors

### LINKED ISSUE #848: "No rule found for" errors are bad
If I create a rule that attempts to invoke an executable that doesn't exist I will get an error like:

```
No rule found for foo/bar.exe
```

This error should really include the loc where this target is required.

### PR DESCRIPTION
**Pull request by rgrinberg**
There was a missing branch (probably accidentally omitted).

Fix #848

### CODE REVIEW

**Comment by rgrinberg** [comment]
And the last commit adds locs for errors caused by setting up the install rules. Unfortunately, the output of the errors is still crappy, as we don't have a good way to display error messages with 2 locations. But it's already an improvement.

File: test/blackbox-tests/test-cases/dune-ppx-driver-system/run.t [hunk 1]
```diff
@@ -142,7 +151,10 @@ Test using installed drivers
   --impl
   driveruser.ml
   --as-ppx
-  Error: Rule failed to generate the following targets:
+  File "dune", line 5, characters 13-57:
+  5 |  (preprocess (pps -arg1 replacesdriver -arg2 -- -foo bar)))
+                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+  Error: Error: Rule failed to generate the following targets:
```
Comment (emillon) [inline test/blackbox-tests/test-cases/dune-ppx-driver-system/run.t]: This displays "Error: Error" here.

Comment (rgrinberg) [reply test/blackbox-tests/test-cases/dune-ppx-driver-system/run.t]: Thanks. Fixed.

**Review by emillon — APPROVED** [review]
Great! In certain situations it might be confusing, but at least if there's a loc we can improve it.

**Pull request merged**